canvas_sdk/value_set/v2022/adverse_event.py

```diff
@@ -0,0 +1,33 @@
+from ..value_set import ValueSet
+
+
+class StatinAllergen(ValueSet):
```

Comment:
This is currently the way the ones in the workflow kit and home-app are structured. These were taken directly from the workflow kit.

Reply:
Yeah, I thought that might be the answer.

I can't decide how I feel about this. Is the existing pattern a weird way to represent what are essentially a set of constants? Usually you'd use a class definition to define the structure of something, and then instances of the class to represent concrete manifestations of that class.

Jumping out of the health care example for a moment, if you wanted to have a bunch of sets that represent the country codes of the world by continent, would we do this:

```python
class NorthAmerica:
    codes = {"US", "CA", ... }

class Europe:
    codes = { ... }
```

Or this:

```python
@dataclass(frozen=True)
class Continent:
    name: str
    codes: set[str]

NORTH_AMERICA = Continent(name="North America", codes={"US", "CA", ...})
EUROPE = Continent(...)
```

I'm thinking myself about whether the existing pattern is Pythonic.

Reply:
@andhines might have good insight here

Reply:
Tradeoffs everywhere but having the value sets in code avoids disk lookup with frequent computations, which is the expectation. That's the main reason to avoid "instances of that class that populate the data" if I'm interpreting that correctly. Other aspect is autocomplete, developer ergonomics of autocompleting the `Diabetes` or `Antihypertensive` valuset in the IDE, which wouldn't be possible otherwise. These classes also support combination via special operator methods as you can see https://github.com/canvas-medical/canvas-workflow-kit/blob/main/canvas_workflow_kit/value_set/value_set.py#L5

The classes also enforce known code systems, better safety versus a dictionary. It seems clear to me the value set thinking was good, still is good, and we should keep it so filtering can be done in memory when useful, instead of having to go to disk over network. 

The additional complexity from recent conversations is how to interpret value sets when they're used in a data module query that's going to disk anyway, and we want to do an in-database join to filter without passing all the literals as part of the query. Seems we should be able to do that with a value set identifier to use in a `WHERE` clause with the join.

Reply:
To be clear, the example above would also be in-memory -- no disk involved. It's just a slightly different way of modeling the data. You would still be able to have the autocompletion, operator methods, enforcement of known code systems, etc.

But if the current way works, it works -- it was just different than I expected. I'll post again if I notice or think of any concrete advantage that could be gained from using a different approach.